canal/dump.go

```diff
@@ -113,6 +114,20 @@ func (h *dumpParseHandler) Data(db string, table string, values []string) error
 	return h.c.eventHandler.OnRow(events)
 }
 
+func (h *dumpParseHandler) Table(db string, query string) error {
+	if err := h.c.ctx.Err(); err != nil {
+		return err
+	}
+
+	event := &replication.QueryEvent{
+		Schema: []byte(db),
+		Query:  []byte(query),
+	}
+	pos := mysql.Position{Name: h.name, Pos: uint32(h.pos)}
+
+	return h.c.eventHandler.OnDDL(nil, pos, event)
```

Comment:
I also suppose we should provide a valid header, but if we can't set proper values, we should at least add detailed comments to the interface